packages/runtime/container-runtime/src/gc/gcConfigs.ts

```diff
@@ -176,21 +169,15 @@ export function generateGCConfigs(
 	});
 
 	const throwOnInactiveLoad: boolean | undefined = createParams.gcOptions.throwOnInactiveLoad;
-	const tombstoneEnforcementAllowed = shouldAllowGcTombstoneEnforcement(
-		createParams.metadata?.gcFeatureMatrix?.tombstoneGeneration /* persisted */,
-		createParams.gcOptions[gcTombstoneGenerationOptionName] /* current */,
-	);
 
 	const throwOnTombstoneLoadConfig =
 		mc.config.getBoolean(throwOnTombstoneLoadOverrideKey) ??
 		createParams.gcOptions[gcDisableThrowOnTombstoneLoadOptionName] !== true;
 	const throwOnTombstoneLoad =
-		throwOnTombstoneLoadConfig &&
-		tombstoneEnforcementAllowed &&
-		!createParams.isSummarizerClient;
+		throwOnTombstoneLoadConfig && shouldRunSweep && !createParams.isSummarizerClient;
```

Comment:
From my understanding and from the PR description, this should check for `sweepEnabled` not `shouldRunSweep`.
Here's my understanding for the different configs for tombstone and sweep:
1. `tombstoneMode` (marking objects as tombstones) is on by default. It can be disabled via feature flags.
2. `sweepEnabled` indicates that both phases of sweep (tombstone and delete) are eligible for a container. `gcGeneration` is used to figure this out.
3. If `sweepEnabled` is true, `throwOnTombstoneLoadConfig` tells whether tombstone load should throw or not.
4. If `shouldRunSweep` is enabled, delete is enabled for the session. `this.configs.enableGCSweep` tells whether sweep should run.

Reply:
You're right.  I got nervous when I realized that the default behavior for (3) is `true` to throw on tombstone load.  But we need to distinguish between that and Delete, and we intentionally made that change.  So I was wrong.